Error when importing imageio source

/kind bug

**What happened**:
importer pod could not import imageio source

**How to reproduce it (as minimally and precisely as possible)**:
Was imported with the following DV

```yaml
apiVersion: cdi.kubevirt.io/v1alpha1
kind: DataVolume
metadata:
  creationTimestamp: '2020-04-21T08:58:17Z'
  generation: 9
  name: 6e79a7a9-c4e2-41ce-a857-9d992c55298b
  namespace: default
  ownerReferences:
    - apiVersion: v2v.kubevirt.io/v1alpha1
      blockOwnerDeletion: true
      controller: true
      kind: VirtualMachineImport
      name: vm-import-win-wgkp6
      uid: 4aa94e10-365b-4930-ae36-2e5d5fcc8065
    - apiVersion: kubevirt.io/v1alpha3
      blockOwnerDeletion: true
      controller: false
      kind: VirtualMachine
      name: win
      uid: d345e977-f004-4530-808e-5ffa4c7cc21b
  resourceVersion: '518199'
  selfLink: >-
    /apis/cdi.kubevirt.io/v1alpha1/namespaces/default/datavolumes/6e79a7a9-c4e2-41ce-a857-9d992c55298b
  uid: a00178d1-a557-4d73-bc6a-1f2e0cdd8b2e
spec:
  pvc:
    accessModes:
      - ReadWriteOnce
    resources:
      requests:
        storage: 1Gi
  source:
    imageio:
      certConfigMap: vmimport.v2v.kubevirt.iog2ghg
      diskId: 6e79a7a9-c4e2-41ce-a857-9d992c55298b
      secretRef: vmimport.v2v.kubevirt.iophjkt
      url: 'https://10-37-140-88.rhev.lab.eng.brq.redhat.com:8443/ovirt-engine/api'
status:
  phase: ImportInProgress
  progress: N/A
  restartCount: 5
```

**Anything else we need to know?**:
- I checked the disk just before the import and it was in OK state (not locked).
- Other disk from that env (same storage domain) was imported successfully 

importer pod logs:
```
I0421 08:58:34.853826       1 importer.go:51] Starting importer
I0421 08:58:34.853956       1 importer.go:108] begin import process
I0421 08:58:36.343101       1 http-datasource.go:218] Attempting to get certs from /certs/ca.pem
I0421 08:58:36.556175       1 data-processor.go:275] Calculating available size
I0421 08:58:36.558254       1 data-processor.go:283] Checking out file system volume size.
I0421 08:58:36.558295       1 data-processor.go:287] Request image size not empty.
I0421 08:58:36.558316       1 data-processor.go:292] Target size 1Gi.
I0421 08:58:36.558626       1 data-processor.go:205] New phase: TransferDataFile
I0421 08:58:36.560969       1 util.go:144] Writing data...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x118e249]

goroutine 61 [running]:
kubevirt.io/containerized-data-importer/pkg/util/prometheus.(*ProgressReader).updateProgress(0x0, 0xc000048990)
	pkg/util/prometheus/prometheus.go:59 +0x29
kubevirt.io/containerized-data-importer/pkg/util/prometheus.(*ProgressReader).timedUpdateProgress(0x0)
	pkg/util/prometheus/prometheus.go:54 +0x3f
created by kubevirt.io/containerized-data-importer/pkg/util/prometheus.(*ProgressReader).StartTimedUpdate
	pkg/util/prometheus/prometheus.go:46 +0x3f
```


importer pod logs in CrashLoopBackOff:

```
I0421 09:03:29.294689       1 importer.go:51] Starting importer
I0421 09:03:29.294890       1 importer.go:108] begin import process
E0421 09:03:29.809532       1 importer.go:124] Fault reason is "Operation Failed". Fault detail is "[Cannot transfer Virtual Disk: The following disks are locked: win_Disk1. Please try again in a few minutes.]". HTTP response code is "409". HTTP response message is "409 Conflict".
Error sending transfer image request
kubevirt.io/containerized-data-importer/pkg/importer.getTransfer
	pkg/importer/imageio-datasource.go:256
kubevirt.io/containerized-data-importer/pkg/importer.createImageioReader
	pkg/importer/imageio-datasource.go:173
kubevirt.io/containerized-data-importer/pkg/importer.NewImageioDataSource
	pkg/importer/imageio-datasource.go:55
main.main
	cmd/cdi-importer/importer.go:122
runtime.main
	GOROOT/src/runtime/proc.go:203
runtime.goexit
	src/runtime/asm_amd64.s:1357
```

**Environment**:
- CDI version: v1.16.0 (also tried  v1.15.0)
- Kubernetes version:
```js
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0", GitCommit:"9e991415386e4cf155a24b1da15becaa390438d8", GitTreeState:"archive", BuildDate:"2020-04-19T14:21:51Z", GoVersion:"go1.14.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.2", GitCommit:"aa10b5b", GitTreeState:"clean", BuildDate:"2020-03-16T18:11:23Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
```

--- a/pkg/importer/format-readers.go
+++ b/pkg/importer/format-readers.go
@@ -262,5 +262,7 @@ func (fr *FormatReaders) Close() (rtnerr error) {
 
 // StartProgressUpdate starts the go routine to automatically update the progress on a set interval.
 func (fr *FormatReaders) StartProgressUpdate() {
-	fr.progressReader.StartTimedUpdate()
+	if fr.progressReader != nil {
+		fr.progressReader.StartTimedUpdate()
+	}
 }

--- a/pkg/importer/format-readers_test.go
+++ b/pkg/importer/format-readers_test.go
@@ -3,6 +3,7 @@ package importer
 import (
 	"fmt"
 	"io"
+	"io/ioutil"
 	"os"
 	"path/filepath"
 	"strings"
@@ -89,4 +90,14 @@ var _ = Describe("Format Readers", func() {
 		table.Entry("should append io.reader", rdrGz, stringRdr, 3, false),
 		table.Entry("should append io.Multireader", rdrMulti, stringRdr, 3, false),
 	)
+
+	It("should not crash on no progress reader", func() {
+		stringReader := ioutil.NopCloser(strings.NewReader("This is a test string"))
+		testReader, err := NewFormatReaders(stringReader, uint64(0))
+		// Not passing a real string, so the header checking will fail.
+		Expect(err).To(HaveOccurred())
+		Expect(testReader.progressReader).To(BeNil())
+		// This should not crash
+		testReader.StartProgressUpdate()
+	})
 })

